lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -260,8 +260,41 @@ void HWMemSimImpl::generateMemory(HWModuleOp op, FirMemory mem) {
     // Read Logic
     Value rdata = getMemoryRead(b, reg, addr, addMuxPragmas);
     if (!ignoreReadEnableMem) {
-      Value x = b.create<sv::ConstantXOp>(rdata.getType());
-      rdata = b.create<comb::MuxOp>(en, rdata, x, false);
+      size_t addrRange = (2 << (addr.getType().getIntOrFloatBitWidth() - 1));
```

Comment:
Maybe `1 << addr.getType().getIntOrFloatBitWidth()`?

Reply:
oh yes, of course !